runtime/src/lib.rs

```diff
@@ -221,35 +220,54 @@ macro_rules! create_zeitgeist_runtime {
-    AuthorInherent: pallet_author_inherent::{Call, Inherent, Pallet, Storage} = 111,
-    AuthorFilter: pallet_author_slot_filter::{Config, Event, Pallet, Storage} = 112,
-    AuthorMapping: pallet_author_mapping::{Call, Config<T>, Event<T>, Pallet, Storage} = 113,
-
-    // XCM
-    CumulusXcm: cumulus_pallet_xcm::{Event<T>, Origin, Pallet} = 120,
-    DmpQueue: cumulus_pallet_dmp_queue::{Call, Event<T>, Pallet, Storage} = 121,
-    PolkadotXcm: pallet_xcm::{Call, Config, Event<T>, Origin, Pallet, Storage} = 122,
-    XcmpQueue: cumulus_pallet_xcmp_queue::{Call, Event<T>, Pallet, Storage} = 123,
-
-    // Third-party
-    Crowdloan: pallet_crowdloan_rewards::{Call, Config<T>, Event<T>, Pallet, Storage} = 130,
-);
+macro_rules! create_zeitgeist_runtime_with_additional_pallets {
+    ($($additional_pallets:tt)*) => {
+        #[cfg(feature = "parachain")]
+        create_zeitgeist_runtime!(
+            // System
+            ParachainSystem: cumulus_pallet_parachain_system::{Call, Config, Event<T>, Inherent, Pallet, Storage, ValidateUnsigned} = 100,
+            ParachainInfo: parachain_info::{Config, Pallet, Storage} = 101,
+
+            // Consensus
+            ParachainStaking: parachain_staking::{Call, Config<T>, Event<T>, Pallet, Storage} = 110,
+            AuthorInherent: pallet_author_inherent::{Call, Inherent, Pallet, Storage} = 111,
+            AuthorFilter: pallet_author_slot_filter::{Config, Event, Pallet, Storage} = 112,
+            AuthorMapping: pallet_author_mapping::{Call, Config<T>, Event<T>, Pallet, Storage} = 113,
+
+            // XCM
+            CumulusXcm: cumulus_pallet_xcm::{Event<T>, Origin, Pallet} = 120,
+            DmpQueue: cumulus_pallet_dmp_queue::{Call, Event<T>, Pallet, Storage} = 121,
+            PolkadotXcm: pallet_xcm::{Call, Config, Event<T>, Origin, Pallet, Storage} = 122,
+            XcmpQueue: cumulus_pallet_xcmp_queue::{Call, Event<T>, Pallet, Storage} = 123,
+
+            // Third-party
+            Crowdloan: pallet_crowdloan_rewards::{Call, Config<T>, Event<T>, Pallet, Storage} = 130,
+
+            // Others
+            $($additional_pallets)*
+        );
 
-#[cfg(not(feature = "parachain"))]
-create_zeitgeist_runtime!(
-    // Consensus
-    Aura: pallet_aura::{Config<T>, Pallet, Storage} = 100,
-    Grandpa: pallet_grandpa::{Call, Config, Event, Pallet, Storage} = 101,
+        #[cfg(not(feature = "parachain"))]
+        create_zeitgeist_runtime!(
+            // Consensus
+            Aura: pallet_aura::{Config<T>, Pallet, Storage} = 100,
+            Grandpa: pallet_grandpa::{Call, Config, Event, Pallet, Storage} = 101,
+
+            // Others
+            $($additional_pallets)*
+        );
+    }
+}
+
+#[cfg(not(feature = "without-sudo"))]
+create_zeitgeist_runtime_with_additional_pallets!(
+    // Others
+    Sudo: pallet_sudo::{Call, Config<T>, Event<T>, Pallet, Storage} = 150,
 );
 
+#[cfg(feature = "without-sudo")]
+create_zeitgeist_runtime_with_additional_pallets!();
```

Comment:
Why is this required here?

Reply:
I found no way to include that feature requirement inside a macro, so I had to create a new wrapper macro and apply the feature requirement on the whole wrapper macro. Let me know if you know a better approach to this, I already wasted like 20-30 minutes figuring out a nicer approach to that.

Reply:
Sorry, I am out here. Maybe you @maltekliemann or @vivekvpandya ?

Reply:
I also don't have anything better than what @sea212  has already done https://stackoverflow.com/questions/63008781/check-a-feature-flag-in-an-item-defined-in-a-macro